Adding ref assembly generation docs (#1115)
aspnet/Extensions: docs/ReferenceAssemblies.md

Reference assemblies
========================

> Most projects in this solution have `ref` directory next to `src` that contains project and source code for reference assembly.

Reviewer: ```suggestion
Most projects in this repo have a `ref` directory next to their `src` that contains the project and source code for a reference assembly.
```

> Reference assemblies contain public API surface of libraries and are used for ASP.NET Core targeting pack generation.

Reviewer: ```suggestion
Reference assemblies contain the public API surface of libraries and are used for ASP.NET Core targeting pack generation.
```

Reviewer: There must not be any changes to reference assemblies in patch releases.

### When changing public API

> Run `dotnet msbuild /t:GenerateReferenceSource` in projects `src` directory

Reviewer: ```suggestion
Run `dotnet msbuild /t:GenerateReferenceSource` in that project's `src` directory
```

### When adding a new project

> Run `.\eng\scripts\GenerateProjectList.ps1` from repository root and `dotnet msbuild /t:GenerateReferenceSource` in projects `src` directory

Reviewer: ```suggestion
Run `.\eng\scripts\GenerateProjectList.ps1` from the repository root and `dotnet msbuild /t:GenerateReferenceSource` in that project's `src` directory
```

> ### To set project properties in reference assembly project

Reviewer: ```suggestion
### To set project properties in a reference assembly project
```

> `ref.csproj` is automaticaly generated and shouldn't be edited. To set project properties on reference assembly project place `Directory.Build.props` next to it and add properties there.

Reviewer: Why would I need to add project properties to a ref csproj? Do we have any examples?

Reviewer: ```suggestion
`ref.csproj` is automaticaly generated and shouldn't be edited. To set project properties on a reference assembly project place a `Directory.Build.props` next to it and add the properties there.
```

> ### My project doesn't need reference assembly

Reviewer: ```suggestion
### My project doesn't need a reference assembly
```

> Set `<HasReferenceAssembly>false</HasReferenceAssembly>` in implementation (`src`) project and re-run `.\eng\scripts\GenerateProjectList.ps1`.

Reviewer: ```suggestion
Set `<HasReferenceAssembly>false</HasReferenceAssembly>` in the implementation (`src`) project and re-run `.\eng\scripts\GenerateProjectList.ps1`.
```

### Regenerate reference assemblies for all projects

> Run `.\eng\scripts\GenerateReferenceAssemblies.ps1` from repository root.

Reviewer: ```suggestion
Run `.\eng\scripts\GenerateReferenceAssemblies.ps1` from the repository root.
```
